### PR TITLE
Add Proto Definitions for the Artifact API

### PR DESCRIPTION
**Pull request by tgroh**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by tgroh** [comment]
R: @kennknowles @herohde

File: sdks/common/runner-api/src/main/proto/beam_artifact_api.proto [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ */
+
+syntax = "proto3";
+
+package org.apache.beam.runner_api.v1;
+
+option java_package = "org.apache.beam.sdk.common.runner.v1";
+option java_outer_classname = "ArtifactApi";
+
+service ArtifactService {
```
Comment (herohde) [inline sdks/common/runner-api/src/main/proto/beam_artifact_api.proto]: We need to separate the submission side and execution side of this API.

File: sdks/common/runner-api/src/main/proto/beam_artifact_api.proto [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+message GetArtifactRequest {
+  string name = 1;
+}
+
+message ArtifactChunk {
+  bytes data = 1;
+}
+
+message PutArtifactRequest {
```
Comment (herohde) [inline sdks/common/runner-api/src/main/proto/beam_artifact_api.proto]: Maybe a oneof header and chunk? It seems odd to have to include the same metadata with every chunk.

**Comment by tgroh** [comment]
Split the APIs, and split the preparation and submission portions of the Job API.

File: sdks/common/runner-api/src/main/proto/beam_artifact_api.proto [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  // before this call is made.
+  //
+  // Throws error INVALID_ARGUMENT if not all of the members of the manifest are present
+  rpc CommitManifest(CommitManifestRequest) returns (CommitManifestResponse);
+}
+
+// A service to retrieve artifacts for use in a Job.
+service ArtifactRetrievalService {
+  // Get the manifest for the job
+  rpc GetManifest(GetManifestRequest) returns (Manifest);
```
Comment (herohde) [inline sdks/common/runner-api/src/main/proto/beam_artifact_api.proto]: GetManifestResponse

Comment (tgroh) [reply sdks/common/runner-api/src/main/proto/beam_artifact_api.proto]: Fixed.

**Comment by kennknowles** [comment]
LGTM

**Pull request closed without merging**